library/core/src/lib.rs

```diff
@@ -236,6 +236,7 @@
 #![feature(trait_alias)]
 #![feature(transparent_unions)]
 #![feature(try_blocks)]
+#![feature(type_alias_impl_trait)]
```

Comment:
While I think the code change you have here looks fine, I'm inclined to avoid `TAIT`s in `library` until they're stabilized, just in case.

Thoughts?